qt-models/diveplannermodel.cpp

```diff
@@ -1118,17 +1118,20 @@ void DivePlannerPointsModel::updateDiveProfile()
 	if (diveplan.is_empty())
 		return;
 
+	// For calculating variations, we need a copy of the plan. We have to copy _before_
+	// calling plan(), because that adds deco stops.
+	bool computeVariations = isPlanner() && shouldComputeVariations();
+	auto plan_copy = computeVariations ?
+		std::make_unique<struct diveplan>(diveplan) :
+		std::make_unique<struct diveplan>();
```

Comment:
Should this just be `nullptr` if `computeVariations` is `false` - we are not using it in this case.
And the same thing below.

Reply:
Yes, that the default constructor is the same as `nullptr`. But I think then one couldn't use the `? :` operator. I'll make this more readable.